src/tools/tests/manifest2proto-test.ts

```diff
@@ -159,6 +161,145 @@ describe('manifest2proto', () => {
     });
   });
 
+  it('encodes entity type with simple refinement', async () => {
+    const location = {
+      start: {offset: 0, line: 0, column: 0},
+      end: {offset: 0, line: 0, column: 0}
+    };
+    const refinement = Refinement.fromAst(
```

Comment:
I would prefer if you wrote a manifest with the refinement you want, parse it and then pull out a refinement from the particle or recipe. Using this internal fromAst method can be bug prone and you may be making assumptions about how refinements are constructed.